Anthony14FR/Gymbro
# feat: adding emails queue, job, admin invite.

Description : Adding "Invite users" to /users when connected with admin account. Create a Queue, a Job and a Controller. Configuration on redis to work with a worker.

TO-DO : 
- `./vendor/bin/sail down -v `
- `./vendor/bin/sail build --no-cache`
- `make up` or `./vendor/bin/sail up`

Notes : 
- Added db reset and seed + make clear on `make up` command.
- Added `make down` to simplify removing containers
- Removed mailisearch (obsolete)
- Added redis config to run jobs and sending mail queues
- Added horizon to see logs of jogs. Simply go to "http://localhost/horizon"

How to test : 

- [ ] Connect as Admin
- [ ] Click on "Inviter des utilisateurs"
- [ ] Enter valid Email
- [ ] Send it
- [ ] Open email
- [ ] Click on link
- [ ] Assert See /register page (must be logout before)

How to check : 
If you're using Horizon as well, go to : http://localhost/horizon/jobs/completed (see that the job was completed nicely)
Check your inbox to see that the mail was received
If there is an error, in a terminal run : `docker compose logs -f worker`
Try to re-run the container by running `docker compose start worker`

Micro fix on programs : placing a placeholder instead of default value
